### PR TITLE
Port to libnm

### PR DESCRIPTION
**Pull request by lkundrak**
The libnm-glib is depreacted for a long time already.

### CODE REVIEW

**Comment by lkundrak** [comment]
Note that this is probably not completely ready yet. I basically just compile-tested.

I'm not entirely familiar with Vala and I suspect I've could have gotten ownership wrong in some places. I intend to give it some more love and testing but I figured out it might be a good idea to share this early.

File: src/common/Widgets/NMVisualizer.vala [hunk 1]
```diff
@@ -183,7 +181,7 @@ public abstract class Network.Widgets.NMVisualizer : Gtk.Grid {
 
         nm_client.get_active_connections ().foreach ((ac) => {
             if (active_vpn_connection == null && ac.get_vpn ()) {
-                active_vpn_connection = (NM.VPNConnection) ac;
+                active_vpn_connection = (NM.VpnConnection) ac;
```
Comment (jeremypw) [inline src/common/Widgets/NMVisualizer.vala]: Should not be space after close bracket when casting.

**Comment by lkundrak** [comment]
> Code style review only:
> See inline comments

Thank you. I believe I addressed all of them.

> Local vapi files are best put in their own './vapi' folder, not in './src'. Even better shipped in /usr/share/vala but that is a packaging issue I think.

I'm not moving those, because this is rather unrelated to my changes and I have no idea how to tell the build system they're supposed to be in vapi.

As a matter of fact, NetworkManager installs libnm vapi in /usr/share/vapi, just the distro used in the travis chooses to remove it. As for libnma.vapi, we're accepting contributions upstream ;)

> Did you git rm the old .vapi file?

It should be obvious that I did. Anything wrong there?

I've now given this some testing. It seems to work a bit better than the old version in the number of warnings. Tested with wired Ethernet, Wi-Fi, and VPN.

**Comment by tintou** [comment]
looks good here, the .vapi is indeed in the wrong folder but that's out of the scope of this merge request

**Comment by lkundrak** [comment]
Ping?

**Comment by danrabbit** [comment]
Thanks for your PR!

**Pull request merged**